magento/magento2
# REST products update category_ids cannot be removed

Fixed issue with not correct updating products category_ids through /rest/ for custom_attribute { 'category_ids } data.


### Description (*)
Added call of assignProductToCategories method in \Magento\Catalog\Model\ProductRepository::save
in case when a product currently don't have extension attributes.
This is needed to fix a problem with not correct updating category_ids  when try to update the product with /rest/all/V1/products/ through custom_attribute { 'category_ids': '['10127', '10048'] },

### Fixed Issues
magento/magento2#20481: REST products update category_ids cannot be removed


### Manual testing scenarios (*)
1. Open in swagger catalogProductRepositoryV1 and call PUT: /V1/products/{sku} 

2. Use test product with such test case:
    "custom_attributes": [
      {
        "attribute_code": "category_ids",
        "value": [
          "1", "2", "3", "4"
        ]
      }
    ]
![testcase](https://user-images.githubusercontent.com/16721414/52043584-c5ec9500-2548-11e9-9f4c-f40d5ee64d35.png)

Result will be such: all category_ids with provided id's "1", "2", "3", "4" will be assigned to product.

3. Then try to remove some categories:  
    "custom_attributes": [
      {
        "attribute_code": "category_ids",
        "value": [
          "1", "3"
        ]
      }
    ]
Result will be such:  category_ids with provided id's "1", "3" will be assigned to product. 
Other category_ids, that where before, will be unassigned from product, as expected.

So now category_ids are updated correct!


### Contribution checklist (*)
